### PR TITLE
feat: cli-utils

### PR DESCRIPTION
**Pull request by zreigz**

### CODE REVIEW

File: pkg/sync/loop.go [hunk 1]
```diff
@@ -4,14 +4,21 @@ import (
...
 	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/cli-utils/pkg/apply"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+)
+
+const (
+	// The field manager name for the ones agentk owns, see
+	// https://kubernetes.io/docs/reference/using-api/server-side-apply/#field-management
+	fieldManager = "application/apply-patch"
```
Comment (michaeljguarino) [inline pkg/sync/loop.go:21]: we should probably make this plural specific, eg `plrl/application/apply-patch`

File: pkg/sync/status.go [hunk 2]
```diff
@@ -2,110 +2,111 @@ package sync
...
+		statsCollector.Handle(e)
+		switch e.Type {
+		case event.ApplyType:
+			gk := e.ApplyEvent.Identifier.GroupKind
+			name := e.ApplyEvent.Identifier.Name
+			if e.ApplyEvent.Error != nil {
+				msg := fmt.Sprintf("%s apply %s: %s\n", resourceIDToString(gk, name),
+					strings.ToLower(e.ApplyEvent.Status.String()), e.ApplyEvent.Error.Error())
+				if e.ApplyEvent.Status == event.ApplyFailed {
+					err = fmt.Errorf(msg)
```
Comment (michaeljguarino) [inline pkg/sync/status.go:33]: these should be collected and passed as service errors in the updateComponents calls

File: pkg/sync/status.go [hunk 3]
```diff
@@ -2,110 +2,111 @@ package sync
...
-			State:     toStatus(obj),
-			Group:     gvk.Group,
-			Kind:      gvk.Kind,
-			Version:   gvk.Version,
-			Namespace: obj.GetNamespace(),
-			Name:      obj.GetName(),
-			Synced:    true,
-		})
+func fromSyncResult(e event.StatusEvent) *console.ComponentAttributes {
+	return &console.ComponentAttributes{
```
Comment (michaeljguarino) [inline pkg/sync/status.go:73]: I think this is missing the `State` field (which i think you can get from the khealth submodule of cli-utils)

**Comment by michaeljguarino** [comment]
Mostly looks good, just some small details that seem to be missed in the updateStatus callback

File: pkg/sync/status.go (hunk 2 above)
Comment (zreigz) [reply pkg/sync/status.go:33]: this error is passed to service errors. It is passed here:
```
	if err := engine.updateStatus(id, components, errorAttributes("sync", err)); err != nil {
		log.Error(err, "Failed to update service status, ignoring for now")
	}
```

File: pkg/sync/status.go (hunk 3 above)
Comment (zreigz) [reply pkg/sync/status.go:73]: I have been working with this

**Review by floreks — APPROVED** [review]

**Pull request merged**